packages/react-components/react-spinbutton/src/stories/SpinButton/SpinButtonAccessibility.md

```diff
@@ -0,0 +1,100 @@
+
+### Keyboarding
+
+#### Tab Order
+1. Value field
+
+#### States
+1. Rest (Focused)
+2. Editing (Focused and editing)
+
+#### Keyboard State Diagram
+
+| Starting state | Transition | Resulting state |
+| ---------- | ------------ | ------- |
+| Content before spin button | Tab | Rest |
+| Rest  | Tab | Content after spin button |
+| Editing  | Tab | Content after spin button (Value committed) |
+| Editing  | Enter | Rest (Value committed)|
+| Rest  | Any Edit Key (that results in a change) | Editing |
+| Editing  | Any Edit Key | Editing |
+| Content after spin button (Value committed) | Shift + Tab | Rest |
+
+##### Edit keys
+| Edit Key | Result |
+| ---------- | ------------ |
+| Home | First item in defined range |
+| End | Last item in defined range |
+| Up arrow | Increments value higher, based on the step prop (defaulting to 1) |
+| Down arrow | Increments value lower, based on the step prop (defaulting to 1) |
+| Page up | Increments value higher, based on the stepPage prop (defaulting to 1) |
+| Page down | Increments value lower, based on the stepPage prop (defaulting to 1) |
+| Typing valid value | Valid value |
+
+### High contrast mode
+
+Be careful about visibility of the spin button arrows in high contrast mode.
+TODO: Either add specifics about SVGs in high contrast and how to customize the icon color without messing up HCM or (ideally) link to the Icon section on high contrast, when it exists.
+
+### Motion and animation
+
+There is an animation when this element receives focus. This respects reduced-motion media queries.
+
+When a user holds down an edit key or mouse click, there is an animation of spinning numbers to provide a sense of how fast the value is changing. This animation is an interaction cue and therefore is acceptable even when a user has set a preference for reduced motion.
+
+### Semantics
+
+## Known issues
+
+Placeholder: If there are any current known bugs with browsers or AT that impact this component, list them here.
+
+Mouse press quick changes to value are not announced to screen readers.
+
+## Usage
+
+
+
+### When to choose Spin button
+
+SpinButtons allow someone to incrementally adjust a value in small steps.
+
+SpinButtons are a better choice than Input and Input with type: number when clear indication is needed that there are maximum and minimum allowed values.
+
+SpinButtons are a better choice than Slider when there are many valid values and Slider would not provide enough granularity to choose a precise value.
+
+### Required props
+
+Placeholder: If there are any props or other specific work that the author must do to make the default usage of the component accessible, list those here. One example is that all form components must have a label provided by the author. If the component has any internally defined strings that should be localized, list those here.
+
+Open question: How does everything read? Name, role, value? What about when there are prefixes or postfixes?
```

Comment:
How the SpinButton is read depends on the screen reader used but generally it reads (see comment above about known issues):

1. Label
2. "spin button"
3. "value" where "value" is the current displayed value

> What about when there are prefixes or postfixes?

Prefixes, postfixes and other display formatted (e.g., mapping the numbers 1-12 to January-December) is handled by setting the `aria-valuetext` attribute.

How this is read also depends on the screen reader used but a screen reader should read the displayed text consistently with how it would read it in other contexts (i.e., "$1" should be read as "one dollar"). [Example](https://react.fluentui.dev/?path=/docs/components-spinbutton--default#display-value).